pkg/sql/pg_catalog_test.go

```diff
@@ -170,9 +215,396 @@ func rewriteDiffs(t *testing.T, diffs PGCatalogTables, diffsFile string) {
+			trimText := strings.TrimSpace(text)
+
+			// Skips PgCatalog constants (except PgCatalogID) as these will be written from pgConstants slice
+			if strings.HasPrefix(trimText, pgCatalogPrefix) && trimText != pgCatalogIDConstant {
+				continue
+			}
+
+			output.appendString(text)
+			output.appendString("\n")
+
+			if trimText == pgCatalogIDConstant {
+				for _, pgConstant := range pgConstants {
+					output.appendString("\t")
+					output.appendString(pgConstant)
+					output.appendString("\n")
+				}
+			}
+		}
+	})
+}
+
+// fixVtable adds missing table's create table constants
+func fixVtable(t *testing.T, notImplemented PGCatalogTables) {
+	fileName := filepath.Join(vtablePkg, pgCatalogGo)
+
+	// rewriteFile first will check existing create table constants to avoid duplicates
+	rewriteFile(fileName, func(input *os.File, output outputFile) {
+		existingTables := make(map[string]struct{})
+		reader := bufio.NewScanner(input)
+		for reader.Scan() {
+			text := reader.Text()
+			output.appendString(text)
+			output.appendString("\n")
+			createTable := tableFinderRE.FindStringSubmatch(text)
+			if createTable != nil {
+				tableName := createTable[1]
+				existingTables[tableName] = none
+			}
+		}
+
+		for tableName, columns := range notImplemented {
+			if _, ok := existingTables[tableName]; ok {
+				// Table already implemented
+				continue
+			}
+			createTable, err := createTableConstant(tableName, columns)
+			if err != nil {
+				// We can not implement this table as this uses types not implemented
+				t.Log(err)
+				continue
+			}
+			output.appendString(createTable)
+		}
+	})
+}
+
+// fixPgCatalogGo will update pgCatalog.allTableNames, pgCatalog.tableDefs and will add needed virtualSchemas
+func fixPgCatalogGo(notImplemented PGCatalogTables) {
+	allTableNamesText := getAllTableNamesText(notImplemented)
+	tableDefinitionText := getTableDefinitionsText(pgCatalogGo, notImplemented)
+
+	rewriteFile(pgCatalogGo, func(input *os.File, output outputFile) {
+		reader := bufio.NewScanner(input)
+		for reader.Scan() {
+			text := reader.Text()
+			trimText := strings.TrimSpace(text)
+			if trimText == virtualTablePosition {
+				//VirtualSchemas doesn't have a particular place to start we just print it before virtualTablePosition
+				output.appendString(printVirtualSchemas(notImplemented))
+			}
+			output.appendString(text)
+			output.appendString("\n")
+
+			switch trimText {
+			case tableDefsDeclaration:
+				printBeforeClosure(reader, output, tableDefsClosure, tableDefinitionText)
+			case allTableNamesDeclaration:
+				printBeforeClosure(reader, output, allTableNamesClose, allTableNamesText)
+			}
+		}
+	})
+}
+
+// printBeforeClosure will skip all the lines and print `s` text when finds the closure text
+func printBeforeClosure(reader *bufio.Scanner, output outputFile, closure string, s string) {
+	for reader.Scan() {
+		text := reader.Text()
+		trimText := strings.TrimSpace(text)
+
+		if strings.HasPrefix(trimText, "//") {
```

Comment:
Can you leave a comment or add an example about why we need this branch

Reply:
Done.